### PR TITLE
chore: run tests against latest Docker engine, nightly

### PR DESCRIPTION
**Pull request by mdelapenya**
- **feat: run tests against latest moby, nightly**
- **revert: do not merge this commit**
- **revert: do not merge this commit**

## What does this PR do?
It adds a GH workflow that runs nightly at 23:59 CEST, running the core tests against the TEST channel of the Docker engine.

## Why is it important?
Detect compatibility issues the soonest.

### CODE REVIEW

**Comment by mdelapenya** [comment]
Important to revert the two last commits 😄

File: .github/workflows/docker-moby-latest.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Tests agains Latest Docker Moby
+
+on:
+  pull_request:
+    types: [opened, reopened]
```
Comment (kiview) [inline .github/workflows/docker-moby-latest.yml:5]: I think we don't need it on individual PRs.

Comment (mdelapenya) [reply .github/workflows/docker-moby-latest.yml:5]: Indeed, that's why it's part of the revert commit 😅

**Comment by mdelapenya** [comment]
Closing in favour of #2566

**Pull request closed without merging**